bricks/basic/src/utils/PubSub.ts

```diff
@@ -0,0 +1,43 @@
+import _ from "lodash";
+
+export class PubSub {
+  #topics: Record<string, any> = {};
+  #subUid = 0;
+
+  publish(topic: string, body: any): void {
+    if (!this.#topics[topic]) {
+      return;
+    }
+    const subscribers = this.#topics[topic];
+    let len = subscribers ? subscribers.length : 0;
+    while (len--) {
+      subscribers[len].func(topic, body);
+    }
+  }
+
+  subscribe(topic: string, fn: (topic: string, detail: any) => void): string {
+    if (!this.#topics[topic]) {
+      this.#topics[topic] = [];
+    }
+    const token = (++this.#subUid).toString();
+    this.#topics[topic].push({
+      token: token,
+      func: fn,
+    });
+    return token;
+  }
+
+  unsubscribe(token: string): void {
+    for (const m in this.#topics) {
+      if (this.#topics[m]) {
+        for (let i = 0, j = this.#topics[m].length; i < j; i++) {
+          if (this.#topics[m][i].token === token) {
+            _.remove(this.#topics[m], function (n) {
+              return n === i;
+            });
+          }
+        }
```

Comment:
好像不对吧，你上面 remove 收到的 `n` 其实不是序号，而是具体的 topic 项。

应该是这个意思？

```suggestion
        _.remove(this.#topics[m], item => item.token === token);
```